src/include/com/amazonaws/kinesis/video/cproducer/Include.h

```diff
@@ -241,6 +241,20 @@ struct __ApiCallbacks {
 };
 typedef struct __ApiCallbacks* PApiCallbacks;
 
+/**
+ * Type of caching implementation
+ */
+typedef enum {
+    // No caching is specified, will make API calls
+    API_CALL_CACHE_TYPE_NONE,
+
+    // The stream should be pre-existing as DescribeStream API call will be emulated
+    API_CALL_CACHE_TYPE_ENDPOINT_ONLY,
+
+    // Call the APIs and cache the result
+    API_CALL_CACHE_TYPE_ALL,

```

Comment:
Let me add it in the comments. ALL is the same as NONE for the first call. It caches the info after the call. While, the endpoint only emulates the call always for non-endpoint API calls.

Reply:
Got it. 
IMO, it would be much simpler for SDK users if we they don't have to understand the underlying behavior and can just focus on the things they care about. So, instead of having modes to indicate API caching which is internal detail, the API can just differentiate between whether the customer has pre-created streams or not.

Reply:
True, this was what I thought too. The problem is that we have multiple public API entry points. We have a choice to either add another boolean to the public APIs (which is very poor API design at that stage) or permute the APIs by creating another set of APIs which would encompass the behavior. I think this would be the least of all evil.